tests/pipeline/test_python_component.py

```diff
@@ -0,0 +1,32 @@
+import dask.dataframe as dd
+import pandas as pd
+import pyarrow as pa
+from fondant.component import DaskLoadComponent
+from fondant.pipeline import Pipeline, lightweight_component
+
+
+def test_lightweight_component():
+    pipeline = Pipeline(name="dummy-pipeline", base_path="./data")
+
+    @lightweight_component(
+        base_image="python:3.8-slim-buster",
+        extra_requires=["pandas", "dask"],
+    )
+    class CreateData(DaskLoadComponent):
+        def __init__(self, **kwargs):
+            pass
+
+        def load(self) -> dd.DataFrame:
+            df = pd.DataFrame(
+                {
+                    "x": [1, 2, 3],
+                    "y": [4, 5, 6],
+                },
+                index=pd.Index(["a", "b", "c"], name="id"),
+            )
+            return dd.from_pandas(df, npartitions=1)
```

Comment:
I would try to add a validation for the return type as well. If a user don't returns a dask or pandas dataframe in the custom components, it will not work. It's not related to your code, we have the same requirement already for the custom components. But a here we are lowering the entry barrier and I can imagine that people will not read the documentation in detail anymore. 

So I think it might be good if we could catch somehow if it is a wrong return type. I guess it should be possible within the decorator implementation as well.